integreat_cms/static/src/js/pages/machine-translation-overlay.ts

```diff
@@ -0,0 +1,241 @@
+/* eslint no-magic-numbers: ["error", { "ignore": [0, 1, 3, 4, 5, 15] }] */
+import { getCsrfToken } from "../utils/csrf-token";
+
+const toggleMachineTranslationForPagesButton = () => {
+    // Only activate button if budget is enough
+    const budget = document.getElementById("machine-translation-overlay-budget-remains-result")
+        .innerText as any as number;
+    const MachineTranslationButton = document.getElementById(
+        "machine-translation-overlay-bulk-action-execute"
+    ) as HTMLButtonElement;
+    if (budget > 0 && document.getElementById("machine-translation-overlay-pages").children.length > 0) {
+        MachineTranslationButton.disabled = false;
+    } else {
+        MachineTranslationButton.disabled = true;
+    }
+};
+
+const getSelectedItems = () => {
+    const checkboxes = document.querySelectorAll(".bulk-select-item");
+    const selectedPages: Array<Element> = [];
+    checkboxes.forEach((checkbox) => {
+        if ((checkbox as HTMLInputElement).checked) {
+            selectedPages.push(checkbox.closest("tr"));
+        }
+    });
+    return selectedPages;
+};
+
+const getPagesAndHixValueForPages = async (url: string) => {
+    let result;
+    await fetch(url, {
+        method: "POST",
+        headers: {
+            "X-CSRFToken": getCsrfToken(),
+        },
+    })
+        .then((response) => response.json())
+        .then((json) => {
+            result = json;
+        })
+        .then();
+    return result;
+};
+
+const setRemainingBudget = () => {
+    const currentBudget = document.getElementById("machine-translation-overlay-current-budget-result")
+        .innerText as any as number;
+    const usageOfBudget = document.getElementById("machine-translation-overlay-budget-usage-result")
+        .innerText as any as number;
+    const remainingBudget = currentBudget - usageOfBudget;
+    document.getElementById("machine-translation-overlay-budget-remains-result").innerHTML =
+        remainingBudget as unknown as string;
+    toggleMachineTranslationForPagesButton();
+};
+
+const setRequiredBudget = (sum: number) => {
+    document.getElementById("machine-translation-overlay-budget-usage-result").innerHTML = "";
```

Comment:
```suggestion
    document.getElementById("machine-translation-overlay-budget-usage-result").textContent = "";
```
I'd suggest to use `textContent` instead of `innerHTML` whenever you're not really inserting raw HTML. This also applies to all other occurrences of `innerHTML`.

Reply:
I tried to apply this as often as I thought it made sense. I keep innerHTML on two occasions on which it serves as a cleaner to get rid off the HTML child elements. I hope that decision was correct